File: tasking/sources/task_dep.1.f90
! @@name:	task_dep.1
! @@type:	F-free
! @@operation:	run
! @@expect:	success
! @@version:	omp_4.0
program example
   integer :: x
   x = 1
   !$omp parallel
   !$omp single
      !$omp task shared(x) depend(out: x)
         x = 2
      !$omp end task
      !$omp task shared(x) depend(in: x)
         print*, "x = ", x
      !$omp end task
   !$omp end single
   !$omp end parallel
end program

File: tasking/sources/task_dep.1.c
/*
* @@name:	task_dep.1
* @@type:	C
* @@operation:	run
* @@expect:	success
* @@version:	omp_4.0
*/
#include <stdio.h>
int main() {
   int x = 1;
   #pragma omp parallel
   #pragma omp single
   {
      #pragma omp task shared(x) depend(out: x)
         x = 2;
      #pragma omp task shared(x) depend(in: x)
         printf("x = %d\n", x);
   }
   return 0;
}
